IdentityCore/src/MSIDError.h

```diff
@@ -41,77 +41,95 @@ extern NSString *MSIDKeychainErrorDomain;
 
 typedef NS_ENUM(NSInteger, MSIDErrorCode)
 {
+    /*! =================================================
+     General Errors (510xx, 511xx)
+     =================================================  */
+    // General internal errors that do not fall into one of the specific type
+    // of an error described below.
     MSIDErrorInternal = -51000,
-    MSIDErrorInvalidInternalParameter = -51001,
     
-    MSIDErrorInvalidDeveloperParameter = -51002,
-    MSIDErrorAmbiguousAuthority     = -51003,
-    MSIDErrorInteractionRequired    = -51004,
+    // Parameter errors
+    MSIDErrorInvalidInternalParameter   = -51101,
+    MSIDErrorInvalidDeveloperParameter  = -51102,
+   
+    // Unsupported functionality
+    MSIDErrorUnsupportedFunctionality = -51199,
     
-    MSIDErrorCacheMultipleUsers     = -51005,
+    /*!
+    =================================================
+     Cache Errors   (512xx,
+                     513xx - Keychain)
+    =================================================
+     */
+
+    // Multiple users found in cache when one was intended
+    MSIDErrorCacheMultipleUsers     = -51201,
     
     /*!
      MSID encounted an error when trying to store or retrieve items from
      keychain. Inspect NSUnderlyingError from the userInfo dictionary for
      more information about the specific error. Keychain error codes are
      documented in Apple's <Security/SecBase.h> header file
      */
-    MSIDErrorTokenCacheItemFailure  = -51006,
-    MSIDErrorWrapperCacheFailure    = -51007,
-    MSIDErrorCacheBadFormat         = -51008,
-    MSIDErrorCacheVersionMismatch   = -51009,
+    MSIDErrorTokenCacheItemFailure  = -51301,
+    MSIDErrorWrapperCacheFailure    = -51302,
+    MSIDErrorCacheBadFormat         = -51303,
+    MSIDErrorCacheVersionMismatch   = -51304,
     
-    MSIDErrorServerInvalidResponse = -51010,
-    MSIDErrorDeveloperAuthorityValidation = -51011,
-    MSIDErrorServerRefreshTokenRejected = -51012,
-    MSIDErrorServerOauth = -51013,
-    MSIDErrorInvalidRequest = -51014,
-    MSIDErrorInvalidClient = -51015,
-    MSIDErrorInvalidGrant = -51016,
-    MSIDErrorInvalidScope = -51017,
-    MSIDErrorInvalidParameter = -51018,
+    /*!
+     =================================================
+     Server errors  (514xx)
+     =================================================
+     */
+    // Server returned a response indicating an OAuth error
+    MSIDErrorServerOauth                = -51401,
+    // Server returned an invalid response
+    MSIDErrorServerInvalidResponse      = -51402,
+    // Server returned a refresh token reject response
+    MSIDErrorServerRefreshTokenRejected = -51403,
+    // Other specific server response errors
+    MSIDErrorInvalidRequest             = -51404,
```

Comment:
nit: should it also have "server" in it?

Reply:
done